--- logwatch-7.3/scripts/services/amavis.pom	2006-01-30 00:52:53.000000000 +0100
+++ logwatch-7.3/scripts/services/amavis	2006-08-29 09:58:07.000000000 +0200
@@ -115,7 +115,8 @@
         $Viruses{$Virus}{$From}++;
       }; # if
 
-   } elsif (($FromIP, $Fromspam, $Towards) = ( $ThisLine =~ /^(?:Passed )?(?:SPAM|Blocked SPAM), (?:\[([^\]]*)\] )*[(<]([^>)]*)[)>] -\> [\(\<]([^\>\)]+)[\)\>]/ )) {
+   } elsif (($Ac,$FromIP, $Fromspam, $Towards)
+        = ( $ThisLine =~ /^(Passed |Blocked )?SPAM(?:MY)?,(?: LOCAL)? (?:\[([^\]]*)\] )*[(<]([^>)]*)[)>] -\> [\(\<]([^\>\)]+)[\)\>]/ )) {
       $SpamMsgs++;
 
       if ($Fromspam =~ /^$/) {
